server/datastore/mysql/software.go

```diff
@@ -4322,6 +4333,10 @@ func (ds *Datastore) ListHostSoftware(ctx context.Context, host *fleet.Host, opt
 				}
 			}
 
+			if icon, ok := iconsBySoftwareTitleID[softwareTitleRecord.ID]; ok {
+				softwareTitleRecord.IconUrl = ptr.String(icon.IconUrl())
```

Comment:
This is incorrect, when we are requesting this from the my device page we need to return 
`device/{token}/software/titles/{software_title_id}/icon`
However, when we are requesting this from the non-my device pages we need to return 
`/api/latest/fleet/software/titles/{software_title_id}/icon?team_id={team_id}`